internals/src/hex/buf_encoder.rs

```diff
@@ -8,10 +8,32 @@ pub use out_bytes::OutBytes;
 
 use super::Case;
 
+/// Trait for types that can be soundly converted to `OutBytes`.
+///
+/// To protect the API from future breakage this sealed trait guards which types can be used with
+/// the `Encoder`. Currently it is implemented for byte arrays of various interesting lengths.
+///
+/// ## Safety
+///
+/// This is not `unsafe` yet but the `as_out_bytes` should always return the same reference if the
+/// same reference is supplied. IOW the returned memory address and length should be the same if
+/// the input memory address and length are the same.
+///
+/// If the trait ever becomes `unsafe` this will be required for soundness.
+pub trait AsOutBytes: out_bytes::Sealed {
+    /// Performs the conversion.
+    fn as_out_bytes(&self) -> &OutBytes;
+
+    /// Performs the conversion.
+    fn as_mut_out_bytes(&mut self) -> &mut OutBytes;
+}
+
 /// Implements `OutBytes`
 ///
 /// This prevents the rest of the crate from accessing the field of `OutBytes`.
 mod out_bytes {
```

Comment:
I don't understand this comment (and module use), the `OutBytes` inner array is private so cannot be accessed by other modules anyways? Or am I missing something? I would have expected to see `OutBytes` as part of `buf_encoder` module and `private::Sealed`.

Reply:
If it was in `buf_encoder` it'd make the field accessible to the rest of the module which we don't want because it raises auditing cost. This is not critical now but if we ever switch to `MaybeUninit` we won't forget about it and will have a much smaller diff. :)

Reply:
Cool, can we change it to "This prevents the rest of the module from accessing the field of `OutBytes`" please.

Reply:
nit only, of course :)

Reply:
and the crate :)

I'd prefer not to mess with a PR that has two ACKs for something this trivial.